[BUG]  array_test.py::test_array_transform_non_deterministic failed with non-UTC time zone

**Describe the bug**
```
 FAILED ../../src/main/python/array_test.py::test_array_transform_non_deterministic[DATAGEN_SEED=1702503689, INJECT_OOM] - galArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
 Project [transform(sequence(0, (cast((rand(5) * 10.0) as int) + 1), None, Some(Iran)), lambdafunction((lambda x#4689 * 22),  AS t#4688]
 +- GpuColumnarToRow false
    +- GpuRange (0, 1, step=1, splits=2)
 FAILED ../../src/main/python/array_test.py::test_array_transform_non_deterministic_second_param[DATAGEN_SEED=1702503689] - galArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
 Project [transform(sequence(0, (cast((rand(5) * 10.0) as int) + 1), None, Some(Iran)), lambdafunction((lambda x#4700 +  x#4700, lambda i#4701, false)) AS t#4699]
 +- GpuColumnarToRow false
    +- GpuRange (0, 1, step=1, splits=2)
```

------------------------------------

```
 =================================== FAILURES ===================================
 ____________________ test_array_transform_non_deterministic ____________________
 [gw2] linux -- Python 3.9.18 /opt/conda/bin/python
 
     def test_array_transform_non_deterministic():
 >       assert_gpu_and_cpu_are_equal_collect(
                 lambda spark : spark.range(1).selectExpr("transform(sequence(0, cast(rand(5)*10 as int) + 1), x -> x * 22) 
                 conf={'spark.rapids.sql.castFloatToIntegralTypes.enabled': True})
 
 ../../src/main/python/array_test.py:336: 
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 ../../src/main/python/asserts.py:588: in assert_gpu_and_cpu_are_equal_collect
     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first, nc_before_compare=result_canonicalize_func_before_compare)
 ../../src/main/python/asserts.py:494: in _assert_gpu_and_cpu_are_equal
     from_gpu = run_on_gpu()
 ../../src/main/python/asserts.py:487: in run_on_gpu
     from_gpu = with_gpu_session(bring_back, conf=conf)
 ../../src/main/python/spark_session.py:122: in with_gpu_session
     return with_spark_session(func, conf=copy)
 ../../src/main/python/spark_session.py:89: in with_spark_session
     ret = func(_spark)
 ../../src/main/python/asserts.py:205: in <lambda>
     bring_back = lambda spark: limit_func(spark).collect()
 ../../../spark-3.1.1-bin-hadoop3.2/python/pyspark/sql/dataframe.py:677: in collect
     sock_info = self._jdf.collectToPython()
 /home/jenkins/agent/workspace/jenkins-rapids_it-non-utc-dev-12/jars/spark-3.1.1-bin-hadoop3.2/python/lib/py4j-0.10.9-way.py:1304: in __call__
     return_value = get_return_value(
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 
 a = ('xro42497', <py4j.java_gateway.GatewayClient object at 0x7f6caea38430>, 'o42496', 'collectToPython')
 kw = {}
 converted = IllegalArgumentException('Part of the plan is not columnar class xecution.ProjectExec\nProject [...:79)\n\tat py4j.GatewayConnection.run(GatewayConnection.java:238)\n\tat hread.java:750)\n', None)
 
     def deco(*a, **kw):
         try:
             return f(*a, **kw)
         except py4j.protocol.Py4JJavaError as e:
             converted = convert_exception(e.java_exception)
             if not isinstance(converted, UnknownException):
                 # Hide where the exception came from that shows a non-Pythonic
                 # JVM exception message.
 >               raise converted from None
 E               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class xecution.ProjectExec
 E               Project [transform(sequence(0, (cast((rand(5) * 10.0) as int) + 1), None, Some(Iran)),  x#4689 * 22), lambda x#4689, false)) AS t#4688]
 E               +- GpuColumnarToRow false
 E                  +- GpuRange (0, 1, step=1, splits=2)
 
 ../../../spark-3.1.1-bin-hadoop3.2/python/pyspark/sql/utils.py:117: IllegalArgumentException
 ----------------------------- Captured stdout call -----------------------------
 ### CPU RUN ###
 ### GPU RUN ###
 _____________ test_array_transform_non_deterministic_second_param ______________
 [gw2] linux -- Python 3.9.18 /opt/conda/bin/python
 
     def test_array_transform_non_deterministic_second_param():
 >       assert_gpu_and_cpu_are_equal_collect(
                 lambda spark : debug_df(spark.range(1).selectExpr("transform(sequence(0, cast(rand(5)*10 as int) + 1), (x, 
                 conf={'spark.rapids.sql.castFloatToIntegralTypes.enabled': True})
 
 ../../src/main/python/array_test.py:341: 
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 ../../src/main/python/asserts.py:588: in assert_gpu_and_cpu_are_equal_collect
     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first, nc_before_compare=result_canonicalize_func_before_compare)
 ../../src/main/python/asserts.py:494: in _assert_gpu_and_cpu_are_equal
     from_gpu = run_on_gpu()
 ../../src/main/python/asserts.py:487: in run_on_gpu
     from_gpu = with_gpu_session(bring_back, conf=conf)
 ../../src/main/python/spark_session.py:122: in with_gpu_session
     return with_spark_session(func, conf=copy)
 ../../src/main/python/spark_session.py:89: in with_spark_session
     ret = func(_spark)
 ../../src/main/python/asserts.py:205: in <lambda>
     bring_back = lambda spark: limit_func(spark).collect()
 ../../src/main/python/array_test.py:342: in <lambda>
     lambda spark : debug_df(spark.range(1).selectExpr("transform(sequence(0, cast(rand(5)*10 as int) + 1), (x, i) -> x + i) 
 ../../src/main/python/data_gen.py:910: in debug_df
     print('COLLECTED\n{}'.format(df.collect()))
 ../../../spark-3.1.1-bin-hadoop3.2/python/pyspark/sql/dataframe.py:677: in collect
     sock_info = self._jdf.collectToPython()
 /home/jenkins/agent/workspace/jenkins-rapids_it-non-utc-dev-12/jars/spark-3.1.1-bin-hadoop3.2/python/lib/py4j-0.10.9-way.py:1304: in __call__
     return_value = get_return_value(
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 
 a = ('xro42665', <py4j.java_gateway.GatewayClient object at 0x7f6caea38430>, 'o42664', 'collectToPython')
 kw = {}
 converted = IllegalArgumentException('Part of the plan is not columnar class xecution.ProjectExec\nProject [...:79)\n\tat py4j.GatewayConnection.run(GatewayConnection.java:238)\n\tat hread.java:750)\n', None)
 
     def deco(*a, **kw):
         try:
             return f(*a, **kw)
         except py4j.protocol.Py4JJavaError as e:
             converted = convert_exception(e.java_exception)
             if not isinstance(converted, UnknownException):
                 # Hide where the exception came from that shows a non-Pythonic
                 # JVM exception message.
 >               raise converted from None
 E               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class xecution.ProjectExec
 E               Project [transform(sequence(0, (cast((rand(5) * 10.0) as int) + 1), None, Some(Iran)),  x#4700 + lambda i#4701), lambda x#4700, lambda i#4701, false)) AS t#4699]
 E               +- GpuColumnarToRow false
 E                  +- GpuRange (0, 1, step=1, splits=2)
 
 ../../../spark-3.1.1-bin-hadoop3.2/python/pyspark/sql/utils.py:117: IllegalArgumentException
 ----------------------------- Captured stdout call -----------------------------
 ### CPU RUN ###
 COLLECTED
 [Row(t=[0, 2])]
 == Physical Plan ==
 Project [transform(sequence(0, (cast((rand(5) * 10.0) as int) + 1), None, Some(Iran)), lambdafunction((lambda x#4694 +  x#4694, lambda i#4695, false)) AS t#4693]
 +- *(1) Range (0, 1, step=1, splits=2)
 
 
 root
  |-- t: array (nullable = true)
  |    |-- element: integer (containsNull = false)
 
 ### GPU RUN ###
```

--- a/integration_tests/README.md
+++ b/integration_tests/README.md
@@ -343,6 +343,24 @@ integration tests. For example:
 $ DATAGEN_SEED=1702166057 SPARK_HOME=~/spark-3.4.0-bin-hadoop3 integration_tests/run_pyspark_from_build.sh
 ```
 
+### Running with non-UTC time zone
+For the new added cases, we should check non-UTC time zone is working, or the non-UTC nightly CIs will fail.
+The non-UTC nightly CIs are verifing all cases with non-UTC time zone.
+But only a small amout of cases are verifing with non-UTC time zone in pre-merge CI due to limited GPU resources.
+When adding cases, should also check non-UTC is working besides the default UTC time zone.
+To run with non-UTC time zone, set TZ environment variable,
+For example:
+```shell
+$ TZ=Iran ./integration_tests/run_pyspark_from_build.sh
+```
+If the new added cases failed with non-UTC, then should allow the operator(does not support non-UTC) fallback,
+For example, add the following annotation to the case:
+```python
+non_utc_allow_for_sequence = ['ProjectExec'] # Update after non-utc time zone is supported for sequence
+@allow_non_gpu(*non_utc_allow_for_sequence)
+test_my_new_added_case_for_sequence_operator()
+```
+
 ### Reviewing integration tests in Spark History Server
 
 If the integration tests are run using [run_pyspark_from_build.sh](run_pyspark_from_build.sh) we have

--- a/integration_tests/README.md
+++ b/integration_tests/README.md
@@ -346,13 +346,15 @@ $ DATAGEN_SEED=1702166057 SPARK_HOME=~/spark-3.4.0-bin-hadoop3 integration_tests
 ### Running with non-UTC time zone
 For the new added cases, we should check non-UTC time zone is working, or the non-UTC nightly CIs will fail.
 The non-UTC nightly CIs are verifing all cases with non-UTC time zone.
-But only a small amout of cases are verifing with non-UTC time zone in pre-merge CI due to limited GPU resources.
+But only a small amout of cases are verifing with non-UTC time zone in the pre-merge CI due to limited GPU resources.
 When adding cases, should also check non-UTC is working besides the default UTC time zone.
-To run with non-UTC time zone, set TZ environment variable,
-For example:
+Please test the following time zones:
 ```shell
 $ TZ=Iran ./integration_tests/run_pyspark_from_build.sh
+$ TZ=America/Los_Angeles ./integration_tests/run_pyspark_from_build.sh
 ```
+`Iran` is non-DST(Daylight Savings Time) time zone and `America/Los_Angeles` is DST time zone.
+
 If the new added cases failed with non-UTC, then should allow the operator(does not support non-UTC) fallback,
 For example, add the following annotation to the case:
 ```python

--- a/integration_tests/src/main/python/array_test.py
+++ b/integration_tests/src/main/python/array_test.py
@@ -17,7 +17,7 @@ import pytest
 from asserts import assert_gpu_and_cpu_are_equal_collect, assert_gpu_and_cpu_are_equal_sql, assert_gpu_and_cpu_error, assert_gpu_fallback_collect
 from data_gen import *
 from conftest import is_databricks_runtime
-from marks import incompat
+from marks import incompat, allow_non_gpu
 from spark_session import is_before_spark_313, is_before_spark_330, is_databricks113_or_later, is_spark_330_or_later, is_databricks104_or_later, is_spark_33X, is_spark_340_or_later, is_spark_330, is_spark_330cdh
 from pyspark.sql.types import *
 from pyspark.sql.types import IntegralType
@@ -332,11 +332,14 @@ def test_array_transform(data_gen):
 
     assert_gpu_and_cpu_are_equal_collect(do_it)
 
+non_utc_allow_for_sequence = ['ProjectExec'] # Update after non-utc time zone is supported for sequence
+@allow_non_gpu(*non_utc_allow_for_sequence)
 def test_array_transform_non_deterministic():
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : spark.range(1).selectExpr("transform(sequence(0, cast(rand(5)*10 as int) + 1), x -> x * 22) as t"),
             conf={'spark.rapids.sql.castFloatToIntegralTypes.enabled': True})
 
+@allow_non_gpu(*non_utc_allow_for_sequence)
 def test_array_transform_non_deterministic_second_param():
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : debug_df(spark.range(1).selectExpr("transform(sequence(0, cast(rand(5)*10 as int) + 1), (x, i) -> x + i) as t")),

